homeassistant/components/izone/climate.py

```diff
@@ -209,7 +209,8 @@ def controller_update(ctrl: Controller) -> None:
                 return
             self.async_write_ha_state()
             for zone in self.zones.values():
-                zone.async_schedule_update_ha_state()
+                if zone.hass is not None:
+                    zone.async_schedule_update_ha_state()
```

Comment:
This is a bad pattern. We should use our dispatch helper to signal the zone entity to update instead. Then we don't need any guard. We should never access the entity api from outside except in our entity helpers.